### DISCUSSION TITLE
Interval

### DISCUSSION THREAD

**Discussion by Raidus** [supabase/supabase#3734] [Questions] [answered]
Hi,

I am trying to adapt following query to supabase-js:

`SELECT * FROM monitoring_bot_slack WHERE sent_at > now() - '2 hour'::interval;`

I tried following in supabase-js

```js
        let { data, error } = await supa
            .from("monitoring_bot_slack")
            .select()
            .filter("sent_at", "gt", `NOW() - '2 hour'::interval`);
```

but getting this error

```js
{
  hint: null,
  message: `invalid input syntax for type timestamp with time zone: "NOW() - '4 days'"`,
  code: '22007',
  details: null
}
```

A hot fix could be creating a view in postgres but what would be the correct syntax in supabase?

**Comment by steve-chavez** [comment, marked as answer]
You can't use expressions as filters, but you can use some special values for timestamps like

```js
 .filter("sent_at", "gt", `now`);
 .filter("sent_at", "gt", `yesterday`);
```

>     .filter("sent_at", "gt", `NOW() - '2 hour'::interval`);

For this case, I'd suggest computing the value from the client side:

```js
var twoHoursBefore = new Date();
twoHoursBefore.setHours(twoHoursBefore.getHours() - 2);

// ...
.filter("sent_at", "gt", twoHoursBefore.toISOString());
```

A view or a sql function is also another option.

**Reply by Raidus** [reply]
ok thanks. I hoped this might be possible with a sql query because then I don't have to take care of timezones. my solution including timezones

```js
        const offset = 1000 * 60 * 60 * 6; //6h days
        const myDate = new Date();
        myDate.setTime(myDate.getTime() - offset);

        const time_string = myDate.toISOString().toLocaleString("de-DE");
```

**Reply by lsei** [reply]
As far as I know `toISOString()` returns a UTC timestamp by default, which is the default timezone for supabase as well, so no need to worry about timezones. :)

**Reply by Electric-Dragon** [reply]
Hi @steve-chavez,
How do I save date in supabase?

I'm currently saving it this way
```
startDate = new Date(startDate); // a string
.insert([{startDate: `${startDate.getFullYear()},${startDate.getMonth()+1},${startDate.getDate()}`, endDate: `${endDate.getFullYear()},${endDate.getMonth()+1},${endDate.getDate()}` }])
```

Is there a better way to store dates?

**Comment by zakaria-chahboun** [comment]
Use RPC function, it's a better solution for me
